src/WebJobs.Extensions.DurableTask.CodeGeneration/WebJobs.Extensions.DurableTask.CodeGen.SourceGenerator/DurableTaskSourceGenerator.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+using WebJobs.Extensions.DurableTask.CodeGeneration.SourceGenerator.Generators;
+
+namespace WebJobs.Extensions.DurableTask.CodeGeneration.SourceGenerator
+{
+    /// <summary>
+    /// Source generator for creating helper classes based on DurableFunction usage within the project.
+    /// </summary>
+    [Generator]
+    public class DurableTaskSourceGenerator : ISourceGenerator
+    {
+        public void Initialize(GeneratorInitializationContext context)
+        {
+            Debug.WriteLine("Initialize source generator");
+            context.RegisterForSyntaxNotifications(() => new DurableTaskSyntaxReceiver());
+        }
+
+        public void Execute(GeneratorExecutionContext context)
+        {
+            try
+            {
+                if (!(context.SyntaxContextReceiver is DurableTaskSyntaxReceiver receiver))
+                    return;
+
+                // Exit early if the receiver did not manage to be initialized
+                if (!receiver.Initialized)
+                    return;
+
+                var parseOptions = context.ParseOptions;
+
+                if (!IGeneratedDurableOrchestrationContextGenerator.TryGenerate(receiver.DurableOrchestrationContextTypeSymbol, out var iOrchestrationContext))
```

Comment:
here starts a chain of if-statements whose purpose is a bit unclear to me. Are we validating the initialization still or what are we testing for? Can we document it?

Reply:
My overall goal is to make sure that if we can't successfully generate everything, we don't generate anything at all. The reasoning behind this is there are certain interdependencies between the things that get generated (i.e. ```IGeneratedDurableOrchestrationContext``` depends on ```IGeneratedDurableActivityCaller```). 

The if statements are where the SourceText gets created for the generated files. If there is any problem generating any of these, I want to skip adding all of them to the context below.

I'll add some comments to this effect to make this clearer.